lib/client/index.js

```diff
@@ -0,0 +1,1710 @@
+          .style('top', (d3.event.pageY + 15) + 'px');
+      })
+      .on('mouseout', function () {
+        tooltip.transition()
+          .duration(TOOLTIP_TRANS_MS)
+          .style('opacity', 0);
+      });
+
+    treatmentDots.append('path')
+      .attr('class', 'path')
+      .attr('fill', function (d) { return d.outlineOnly ? 'transparent' : d.color; })
+      .attr('stroke-width', function (d) { return d.outlineOnly ? 1 : 0; })
+      .attr('stroke', function (d) { return d.color; })
+      .attr('id', function (d, i) { return 's' + i; })
+      .attr('d', arc);
+
+
+    // labels for carbs and insulin
+    if (showValues) {
+      var label = treatmentDots.append('g')
+        .attr('class', 'path')
+        .attr('id', 'label')
+        .style('fill', 'white');
+      label.append('text')
+        .style('font-size', 40 / scale)
+        .style('text-shadow', '0px 0px 10px rgba(0, 0, 0, 1)')
+        .attr('text-anchor', 'middle')
+        .attr('dy', '.35em')
+        .attr('transform', function (d) {
+          d.outerRadius = d.outerRadius * 2.1;
+          d.innerRadius = d.outerRadius * 2.1;
+          return 'translate(' + arc.centroid(d) + ')';
+        })
+        .text(function (d) { return d.element; });
+    }
+  }
+
+  function updateClock() {
+    updateClockDisplay();
+    var interval = (60 - (new Date()).getSeconds()) * 1000 + 5;
+    setTimeout(updateClock,interval);
+
+    updateTimeAgo();
+
+    // Dim the screen by reducing the opacity when at nighttime
+    if (browserSettings.nightMode) {
+      var dateTime = new Date();
+      if (opacity.current !== opacity.NIGHT && (dateTime.getHours() > 21 || dateTime.getHours() < 7)) {
+        $('body').css({ 'opacity': opacity.NIGHT });
+      } else {
+        $('body').css({ 'opacity': opacity.DAY });
+      }
+    }
+  }
+
+  function updateClockDisplay() {
+    if (inRetroMode()) {
+      return;
+    }
+    now = Date.now();
+    $('#currentTime').text(formatTime(new Date(now), true)).css('text-decoration', '');
+  }
+
+  function getClientAlarm(type) {
+    var alarm = clientAlarms[type];
+    if (!alarm) {
+      alarm = { type: type };
+      clientAlarms[type] = alarm;
+    }
+    return alarm;
+  }
+
+  function isTimeAgoAlarmType(alarmType) {
+    return alarmType === 'warnTimeAgo' || alarmType === 'urgentTimeAgo';
+  }
+
+  function checkTimeAgoAlarm(ago) {
+    var level = ago.status
+      , alarm = getClientAlarm(level + 'TimeAgo');
+
+    if (Date.now() >= (alarm.lastAckTime || 0) + (alarm.silenceTime || 0)) {
+      currentAlarmType = alarm.type;
+      console.info('generating timeAgoAlarm', alarm.type);
+      container.addClass('alarming-timeago');
+      var message = {'title': 'Last data received ' + [ago.value, ago.label].join(' ')};
+      if (level === 'warn') {
+        generateAlarm(alarmSound, message);
+      } else {
+        generateAlarm(urgentAlarmSound, message);
+      }
+    }
+  }
+
+  function updateTimeAgo() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/799312596/issues/source?bid=2452379&fileBranchId=2725007#l1243)